### PR TITLE
ENG-519 implement app-builder side for expand/collapse all

### PR DESCRIPTION
**Pull request by ichalagashvili**
- This PR is markes as `draft` because once `frontend-libraries` has a newer version of `pagetreeselector` its version also needs to be bumped up in `app-builder`.

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ichalagashvili**

If they are not already assigned, you can assign the PR to them by writing `/assign @ichalagashvili` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/state/pages/reducer.js [hunk 1]
```diff
@@ -156,6 +158,15 @@ const titlesMap = (state = {}, action = {}) => {
   }
 };
 
+const toggleBatchExpandedValues = (arr, toggleValue) => {
+  const newState = {};
+  arr.map((pg) => {
+    newState[pg] = { loaded: true, loading: false, expanded: toggleValue };
```
Comment (gidesan) [inline src/state/pages/reducer.js:164]: please use object destructuring here

File: src/state/pages/reducer.js [hunk 2]
```diff
@@ -183,6 +194,19 @@ const statusMap = (state = {}, action = {}) => {
...
     }
+    case BATCH_TOGGLE_EXPANDED: {
+      const pageCodes = action.payload;
+      return toggleBatchExpandedValues(pageCodes, true);
+    }
+    case COLLAPSE_ALL: {
+      const newState = {};
+      const pageCodes = Object.keys(state);
+      pageCodes.map((p) => {
+        newState[p] = { expanded: false, loading: false, loaded: state[p].loaded };
```
Comment (gidesan) [inline src/state/pages/reducer.js:205]: please use object destructuring here

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/709#pullrequestreview-414728005" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**